app/services/create_reservation.rb

```diff
@@ -39,22 +42,47 @@ def user_already_has_reservation?
     user_already_has_reservation
   end
 
-  def event_is_at_capacity?
-    event_is_at_capacity = event_reservations.count >= event_capacity
+  def event_date_capacity?
+    event_date_capacity =
+      event_reservations.count >= event_date[:capacity]
 
-    if event_is_at_capacity
+    if event_date_capacity
       reservation.errors.add(:event_date_id, 'is at capacity')
     end
 
-    event_is_at_capacity
+    event_date_capacity
   end
 
-  def event_capacity
-    PantryFinderApi.new.event_date(event_date_id)[:capacity]
+  def event_slot_capacity?
+    return unless event_slot_id
+
+    event_slot_capacity =
+      event_reservations.count >= event_slot[:capacity]
+
+    if event_slot_capacity
+      reservation.errors.add(:event_slot_id, 'is at capacity')
+    end
+
+    event_slot_capacity
+  end
+
+  def event_date
+    PantryFinderApi.new.event_date(event_date_id)
```

Comment:
This value needs to be cached on an instance variable to prevent making duplicate api calls.